apm-sniffer/apm-sdk-plugin/websphere-liberty-23.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/websphereliberty/v23/AsyncContextInterceptor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.websphereliberty.v23;
+
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.plugin.websphereliberty.v23.async.AsyncType;
+import org.apache.skywalking.apm.plugin.websphereliberty.v23.async.RunnableWrapper;
+
+public class AsyncContextInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        Runnable runnable = (Runnable) allArguments[0];
+
+        if (runnable instanceof RunnableWrapper) {
```

Comment:
About this `instanceof`, is this executed more than once?

Reply:
Yes. In websphere liberty, start runnable created by `AsyncContext.start()` method finally will be submit to an executor by `AsyncContext.startWithExecutorThread` method. So enhance startWithExecutorThread is enough.

But in websphere traditional edition, start method has a check, if `WSAsyncContextImpl.wcConfig.isUseAsyncRunnableWorkManager()==true`, start runnable cant not enter startWithExecutorThread method. So we must enhance start method too.
![image](https://github.com/apache/skywalking-java/assets/16346644/56353234-0e73-458f-9ab8-2733d159ce12)

Both enhance start & startWithExecutorThread will cause AsyncContextInterceptor run twice. So I have a instanceof RunnableWrapper check.